### PR TITLE
Added support for the proxy_* params that are supported in clustertask task-maven

### PR DESCRIPTION
**Pull request by Senjuti256**
With reference to issue #2 I have enabled support for all the proxy_* params present in task-maven clustertask. I have enabled it using workspaces and configmaps. 
Signed off by :- @Senjuti256 <sde@redhat.com>

### CODE REVIEW

**Comment by Aneesh-M-Bhat** [comment]
The implementation looks good to me.
/lgtm

**Review by Aneesh-M-Bhat — APPROVED** [review]

File: scripts/maven-generate.sh [hunk 1]
```diff
@@ -32,25 +32,40 @@ EOF
...
+    if test -f ${WORKSPACES_PROXY_SECRET_PATH}/username && test -f ${WORKSPACES_PROXY_SECRET_PATH}/password; then
+    PARAMS_PROXY_USER=$(cat ${WORKSPACES_PROXY_SECRET_PATH}/username)
+    PARAMS_PROXY_PASSWORD=$(cat ${WORKSPACES_PROXY_SECRET_PATH}/password)
+
+    # Fetching proxy configuration values from ConfigMap workspace
+    PARAMS_PROXY_PORT=$(cat ${WORKSPACES_PROXY_CONFIGMAP_PATH}/proxy_port)
+    PARAMS_PROXY_PROTOCOL=$(cat ${WORKSPACES_PROXY_CONFIGMAP_PATH}/proxy_protocol)
+    PARAMS_PROXY_NON_PROXY_HOSTS=$(cat ${WORKSPACES_PROXY_CONFIGMAP_PATH}/proxy_non_proxy_hosts)
+
+    if [ -n "${PARAMS_PROXY_HOST}" -a -n "${PARAMS_PROXY_PORT}" ]; then
```
Comment (savitaashture) [inline scripts/maven-generate.sh:45]: from where are we getting `PARAMS_PROXY_HOST` ?

**Comment by Senjuti256** [comment]
@savitaashture I have made the required changes

File: templates/task-maven.yaml [hunk 2]
```diff
@@ -20,6 +20,12 @@ spec:
     - name: sever_secret
```
Comment (savitaashture) [inline templates/task-maven.yaml:20]: I see name mismatch its `sever_secret` here and everywhere its used `server_secret`

File: Makefile [hunk 3]
```diff
@@ -22,10 +22,14 @@ E2E_TESTS ?= ./test/e2e/*.bats
 
 E2E_PVC ?= test/e2e/resources/pvc-maven.yaml
 E2E_SECRET ?= test/e2e/resources/secret-maven.yaml
+E2E_PROXY_SECRET ?= test/e2e/resources/secret-proxy-maven.yaml                #
```
Comment (savitaashture) [inline Makefile:25]: what is the difference between these 2 secrets ???

Comment (Senjuti256) [reply Makefile:25]: The difference between server_secret and proxy_secret lies in their usage and the type of information they retrieve :-
- server_secret : retrieves server access credentials.
- proxy_secret : retrieves proxy server access credentials and proxy configuration values.

A proxy server can be used to access repositories located in restricted or external networks by routing requests through it. Similar to accessing a repository server, if your network environment requires authentication to access the proxy server, you'll need to provide credentials to Maven. These credentials are also stored in the settings XML file under the <proxies> section.
In Maven, a "server" typically refers to a remote repository server from which Maven retrieves dependencies or to which it deploys artifacts.

**Pull request merged**